### PR TITLE
[numpy] Fix less/greater bug with scalar input

### PR DESCRIPTION
**Pull request by Yiyan66**
## Description ##
1. FFI greater
2. Fix a bug in ufunc op: less/greater/less_equal/great_equal
   described in #18594 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by Yiyan66** [comment]
@mxnet-bot run ci [centos-cpu, unix-cpu]

**Comment by Yiyan66** [comment]
@mxnet-bot run ci [centos-cpu, unix-cpu]

**Comment by Yiyan66** [comment]
@mxnet-bot run ci [centos-cpu, unix-cpu]

File: src/api/operator/numpy/np_elemwise_broadcast_logic_op.cc [hunk 1]
```diff
@@ -44,31 +44,65 @@ MXNET_REGISTER_API("_npi.not_equal")
...
+  int result = 0;
+  if (args[1].type_code() == kNDArrayHandle) {
+    result++;
+    result <<= 1;
+  }
+  if (args[0].type_code() == kNDArrayHandle) {
+    result++;
+  }
+
+  switch (result) {
```
Comment (yzhliu) [inline src/api/operator/numpy/np_elemwise_broadcast_logic_op.cc:59]: this is a clever trick, yet I suggest to explicitly use `if (..&&..) ... else if (.. && ..) ... else` to make it more clear and readable.

File: src/api/operator/numpy/np_elemwise_broadcast_logic_op.cc [hunk 2]
```diff
@@ -44,31 +44,65 @@ MXNET_REGISTER_API("_npi.not_equal")
   UFuncHelper(args, ret, op, op_scalar, nullptr);
 });
 
+void set_UFuncHelper(runtime::MXNetArgs args, runtime::MXNetRetValue* ret,
```
Comment (yzhliu) [inline src/api/operator/numpy/np_elemwise_broadcast_logic_op.cc:47]:
```suggestion
void SetUFuncHelper(runtime::MXNetArgs args, runtime::MXNetRetValue* ret,
```

**Pull request merged**